src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogPage.cs

```diff
@@ -963,7 +972,11 @@ internal void ApplyInitialization()
         ///   Raises the <see cref="Destroyed"/> event.
         /// </summary>
         /// <param name="e">An <see cref="EventArgs"/> that contains the event data.</param>
-        internal protected void OnDestroyed(EventArgs e) => Destroyed?.Invoke(this, e);
+        internal protected void OnDestroyed(EventArgs e)
+        {
+            Destroyed?.Invoke(this, e);
+            Dispose();
```

Comment:
Note that the  `TaskDialogPage.Destroyed` event is currently called when the task dialog is closing. If the page (and therefore the icon) is destroyed here, it would mean you cannot display the same `TaskDialogPage` instance (or `TaskDialogIcon` instance) more than once, e.g.:
```c#
using var bitmap = new Bitmap(32, 32, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
using var myPage = new TaskDialogPage() { Icon = new TaskDialogIcon(bitmap) };

TaskDialog.ShowDialog(myPage);
TaskDialog.ShowDialog(myPage); // fails with COMException
```

Reply:
What if one TaskDialogIcon instance is set into two different TaskDialogPage? Is that supported?

In that case I think TaskDialogIcon should have the same lifetime semantics as, say, an Icon or Font on a Form.

Reply:
Good point @kpreisser.
So leave it up to the user to dispose the page?

Reply:
@weltkante Good point. Yes, currently it's supported to use the same `TaskDialogIcon` instance with different `TaskDialogPage` instances at the same time.

@RussKie Yes, and with @weltkante's comment probably `TaskDialogPage` shouldn't dispose the `TaskDialogIcon` (and therefore wouldn't need to implement `IDisposable`).

Reply:
Updated